### PR TITLE
Web Worker support.

### PR DESCRIPTION
**Pull request by qm3ster**
Checks for presence of browser environment before attempting document.URL

### CODE REVIEW

**Comment by qm3ster** [comment]
Not sure if I did a potato, literally my first time on GitHub.

**Comment by mcollina** [comment]
The CI build failed, can you please resolve the issues?

File: lib/connect/ws.js [hunk 1]
```diff
@@ -22,9 +22,13 @@ function buildBuilder (client, opts) {
 }
 
 function buildBuilderBrowser (mqttClient, opts) {
-  var url,
+  var url, parsed;
+  if (typeof(document) !== 'undefined') { // for Web Workers! P.S: typeof(document) !== undefined may be becoming the faster one these days.
     parsed = _URL.parse(document.URL);
-
+  } else {
+    parsed = 'ws://localhost'; // what should happen if URL is not determinable and not provided?
```
Comment (mcollina) [inline lib/connect/ws.js]: just throw an exception

**Comment by qm3ster** [comment]
Done.
There was also
```
  if (!opts.hostname) {
    opts.hostnme = opts.host;
  }
```
(misspelled hostnme)
I changed that too, hope that's ok.

**Comment by mcollina** [comment]
Thanks!!!

**Pull request merged**